src/index.tsx

```diff
@@ -50,19 +51,24 @@ export default class CsvDownload extends React.Component<ICsvDownloadProps> {
       children, text,
       filename, suffix, prefix, bom,
       columns, datas, separator, noHeader, wrapColumnChar, newLineAtEnd, chunkSize,
+      disabled,
       ...props
     } = this.props
 
+    let onClick = this.handleClick
```

Comment:
I would keep the code in render method as it was. Update just handleClick method to return early, if disabled.

Reply:
@dolezel done :)